BxlAsLib.txt

```diff
@@ -0,0 +1,377 @@
+      - Fragment's dependencies must be encoded into the fragment itself.
+      - See example below for details.
+
+  - Other frontends can consume outputs of fragments. For example, one will write the following in
+    DScript:
+    /**
+     * Assume that the fragment is constructed as follows:
+     *   var f = InitGraphFragment("Word.Debug.X64");
+     *   ...
+     *   f.AddOutputFileValue(S("word.exe"), F("Target/Debug/X64/word.exe", 1));
+     */
+    const fragmentWordDebugX64 = readFragment(f`./word.debug.x64.frag`);
+    const dropOutput = DropSDK.dropFile(fragmentWordDebugX64.word.exe);
+  
+  - Fragment can consume outputs produced by other frontends or by other fragments.
+    - Option 1: Use "readFragment" method
+      /**
+       * Assume that the Word fragment is constructed as follows:
+       *   ---------------------------------
+       *   var f = InitGraphFragment("Word.Debug.X64");
+       *   f.AddProcess(new ProcessData {
+       *       SymbolicDependencies: new[] { 
+       *           new SymbolicDependency { Module: "Liblet.Debug.X64", Id: liblet.dll },
+       *           new SymbolicDependency { Module: "Utils", Id: IO.common.dll }
+       *       } 
+       *   });
+       *   ----------------------------------
+       *
+       * Liblet fragment is constructed as follows:
+       *   ----------------------------------
+       *   var f = InitGraphFragment("Liblet.Debug.X64");
+       *   ...
+       *   f.AddOutputFileValue(liblet.dll, F("Target/Debug/X64/liblet.dll", 1));
+       *   ----------------------------------
+       */
+       const libletFragment = readFragment(
+           f`./liblet.debug.x64.frag`,
+           fragmentDependencies: [],
+           fileDependencies: [],
+           directoryDependencies: []);
+       const wordFragment = readFragment(
+           f`./word.debug.x64.frag`,
+           fragmentDependencies: [libletFragment],
+           fileDependencies: [import("Utils").IO.common.dll]
+       )
+    
+    - Option 2: fragments as another resolver kind.
+        interface IGraphFragment
+        {
+            ...
+            /**
+             * Upon deserialization, this dependency will cause evaluation
+             * to ensure that the value for the dependency is created.
+             * The evaluation will also involve fragment loading if the module
+             * in the symbolic dependency refers to a fragment.
+             * This method should be called at the beginning before adding pips.
+             */ 
+            void AddSymbolicDependency(SymbolicDependency dependency);
+        }
+
+      In DScript, one can use the fragment as follows:
+        const f = import("Word.Debug.X64");
+        const o = DropSDK.dropFile(f.word.exe);
+
+BuildXL's friends
+-----------------
+- The above API is the bare minimal one for creating fragments.
+- Dealing with BuildXL's friends is tricky:
+  - For drop, the user needs to add the finalization pip before its dependencies.

```

Comment:
Since drop is already a "friend", the user doesn't need to specify the finalization command at all, because it is always the same for all drops anyway, so BuildXL can always add it.